modules/javafx.media/src/main/native/jfxmedia/jni/Logger.cpp

```diff
@@ -121,29 +121,27 @@ bool CLogger::init(JNIEnv *pEnv, jclass cls)
 
     if (!m_areJMethodIDsInitialized) {
         jclass local_cls = pEnv->FindClass("com/sun/media/jfxmedia/logging/Logger");
-        if (javaEnv.clearException()) {
+        if (javaEnv.clearException() || NULL == local_cls) {
             return false;
         }
 
-        if (NULL != local_cls) {
-            // Get global reference
-            m_cls = (jclass)pEnv->NewWeakGlobalRef(local_cls);
-            pEnv->DeleteLocalRef(local_cls);
-
-            if (NULL != m_cls) {
-                m_logMsg1Method = pEnv->GetStaticMethodID(m_cls, "logMsg", "(ILjava/lang/String;)V");
-                if (javaEnv.clearException()) {
-                   return false;
-                }
-
-                m_logMsg2Method = pEnv->GetStaticMethodID(m_cls, "logMsg", "(ILjava/lang/String;Ljava/lang/String;Ljava/lang/String;)V");
-                if (javaEnv.clearException()) {
-                   return false;
-                }
-
-                if (NULL != m_logMsg1Method && NULL != m_logMsg2Method) {
-                    m_areJMethodIDsInitialized = true;
-                }
+        // Get global reference
+        m_cls = (jclass)pEnv->NewWeakGlobalRef(local_cls);
+        pEnv->DeleteLocalRef(local_cls);
```

Comment:
What is this code attempting to do? After `local_cls` is deleted, `m_cls` refers to a potentially freed object and is thus not safe to use. In order to do anything meaningful with `m_cls`, a strong reference needs to be acquired with `NewLocalRef(m_cls)`.

Reply:
I think it should be fine to use `NewWeakGlobalRef`. `NewLocalRef` will make reference which is valid only for duration of `CLogger::init` and we using `m_cls` in `CLogger::logMsg` as well. I think you mean `NewGlobalRef`, but in this case there is no way to call `DeleteGlobalRef`, since CLogger instance is static global variable which gets deleted only when native media library gets unloaded and by this time JVM will be shutting down. Java Logger class referenced by m_cls should not be freed, since it contains only static methods and fields, until our native code unloads. So, using weak reference should be fine in this case.